src/main/java/com/minecolonies/coremod/client/render/TileEntityInfoPosterRenderer.java

```diff
@@ -0,0 +1,254 @@
+     */
+    private static final double BLOCK_MIDDLE = 0.5;
+
+    /**
+     * Used to move the text slightly in the front.
+     */
+    private static final double SLIGHTLY_IN_FRONT = 0.01;
+
+    /**
+     * Scaling factor.
+     */
+    private static final double SCALING_FACTOR = 0.010416667F;
+
+    /**
+     * Y-Offset in order to have the scarecrow over ground.
+     */
+    private static final double YOFFSET    = 0.2;
+
+    /**
+     * 90° offset.
+     */
+    private static final int NINETY_DEGREE = 90;
+
+    /**
+     * 180° offset.
+     */
+    private static final int HALF_ROTATION = 180;
+
+    /**
+     * 270° offset.
+     */
+    private static final int THREE_QUARTERS  = 270;
+
+    /**
+     * Max text length.
+     */
+    private static final int MAX_TEXT_LENGTH = 90;
+
+    /**
+     * Text offset at x.
+     */
+    private static final int TEXT_OFFSET_X   = 10;
+
+    /**
+     * Text offset at y.
+     */
+    private static final int TEXT_OFFSET_Y   = 5;
+
+    /**
+     * The ModelSign instance for use in this renderer
+     */
+    private IBakedModel model = null;
+
+    @Override
+    public void renderTileEntityAt(final TileEntityInfoPoster te, final double x, final double y, final double z, final float partialTicks, final int destroyStage)
+    {
+        if (model == null)
+        {
+            model = loadModel();
+        }
+
+        final World world = te.getWorld();
+        final IBlockState state = world.getBlockState(te.getPos());
+        final BlockPos pos = te.getPos();
+        final IBlockState actualState = state.getBlock().getActualState(state, world, pos);
+        int facing = (int) actualState.getValue(BlockWallSign.FACING).getHorizontalAngle();
+
+        double plusX = 0;
+        double plusZ = 0;
+
+        switch(facing)
+        {
+            case NINETY_DEGREE:
+                plusX +=1;
+                plusZ+=1;
+                break;
+            case HALF_ROTATION:
+                plusZ+=1;
+                facing= 0;
+                break;
+            case 0:
+                facing = HALF_ROTATION;
+                plusX += 1;
+                break;
+            default:
+                //do nothing
+
+        }
+
+        GlStateManager.pushMatrix();
+        GlStateManager.translate(x + plusX, y + YOFFSET, z + plusZ);
+        GlStateManager.rotate(facing, 0.0F, 1.0F, 0.0F);
+
+        GlStateManager.disableCull();
+
+        RenderHelper.disableStandardItemLighting();
+
+        GlStateManager.color(1F, 1F, 1F, 1F);
+
+        final int alpha = ((int) (1.0D * 0xFF)) << 24;
+
+        GlStateManager.enableBlend();
+        GlStateManager.enableTexture2D();
+
+        GlStateManager.blendFunc(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA);
+        GlStateManager.colorMask(false, false, false, false);
+        GlStateManager.colorMask(true, true, true, true);
+        GlStateManager.depthFunc(GL11.GL_LEQUAL);
+        renderModel(world, model, pos, alpha);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') "NullPointerException" will be thrown when invoking method "renderModel()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2259)